src/window.rs

```diff
@@ -176,46 +176,44 @@ impl WindowWrapper {
+        let raw_handle = self.window.raw();
 
-            if let Ok(rect) = self.window.subsystem().display_bounds(display_index) {
-                if self.fullscreen {
+        if self.fullscreen {
+            if cfg!(not(target_os = "macos")) {
+                unsafe {
                     // Set window back to resizable
                     sdl2::sys::SDL_SetWindowResizable(raw_handle, sdl2::sys::SDL_bool::SDL_TRUE);
+                }
+            }
 
-                    // Use cached size and position
-                    self.window
-                        .set_size(self.cached_size.0 as u32, self.cached_size.1 as u32)
-                        .unwrap();
-                    self.window.set_position(
-                        sdl2::video::WindowPos::Positioned(self.cached_position.0),
-                        sdl2::video::WindowPos::Positioned(self.cached_position.1),
-                    );
-                    self.window.set_bordered(true);
-                } else {
-                    // Cache the size and position
-                    sdl2::sys::SDL_GetWindowSize(
-                        raw_handle,
-                        &mut self.cached_size.0,
-                        &mut self.cached_size.1,
-                    );
-                    sdl2::sys::SDL_GetWindowPosition(
-                        raw_handle,
-                        &mut self.cached_position.0,
-                        &mut self.cached_position.1,
-                    );
-                    sdl2::sys::SDL_SetWindowResizable(raw_handle, sdl2::sys::SDL_bool::SDL_FALSE);
-
+            // Use cached size and position
+            self.window
+                .set_size(self.cached_size.0, self.cached_size.1)
+                .unwrap();
+            self.window.set_position(
+                sdl2::video::WindowPos::Positioned(self.cached_position.0),
+                sdl2::video::WindowPos::Positioned(self.cached_position.1),
+            );
+        } else {
+            self.cached_size = self.window.size();
+            self.cached_position = self.window.position();
+
+            if cfg!(not(target_os = "macos")) {
+                let raw_handle = self.window.raw();
+                let display_index = self.window.display_index().unwrap();
+                if let Ok(rect) = self.window.subsystem().display_bounds(display_index) {
+                    unsafe {
+                        sdl2::sys::SDL_SetWindowResizable(raw_handle, sdl2::sys::SDL_bool::SDL_FALSE);
+                    }
                     // Set window to fullscreen
                     self.window.set_size(rect.width(), rect.height()).unwrap();
                     self.window.set_position(
                         sdl2::video::WindowPos::Positioned(rect.x()),
                         sdl2::video::WindowPos::Positioned(rect.y()),
                     );
-                    self.window.set_bordered(true);
                 }
+            } else {
+                self.window.maximize();
```

Comment:
So depending on the behavior on window this might need to be changed? does full screen on windows make either
- New desktop
- Or completely cover the whole screen even the taskbar? (and it becomes borderless)

Because if so then macOS has similar thing where it'll move window to a new Space and toggle screen wide full screen

Reply:
`window.maximize()` on the other hand just sets the window to cover all available space minus menubar (ala taskbar)

Just for feature consistency across systems

Reply:
fullscreen covers everything including the taskbar which is the desired functionality. I figured window.maximize() would swap to a new desktop on macos which is the equivalent functionality on that platform. Maybe I'm misunderstanding?

Reply:
@Kethku  no that would be then to achieve this
```rust
self.window.set_fullscreen(FullscreenType::Desktop).ok();
```
However you need to set it back to `Off` when going back to normal so
```rust
self.window.set_fullscreen(FullscreenType::Off).ok();
```

Reply:
I see. I will update to match this. Do you think linux machines would expect that method as well?

Reply:
does that work on window?, what if you use `set_fullscreen(FullscreenType::Desktop)` instead of all the current functionality, SDL docs specify that it should behave across platforms normally so perhaps does the job can you test it?

```rust
pub fn toggle_fullscreen(&mut self) {
        if self.fullscreen {
            self.window.set_fullscreen(FullscreenType::Off).ok();
            self.window
                .set_size(self.cached_size.0, self.cached_size.1)
                .unwrap();
            self.window.set_position(
                sdl2::video::WindowPos::Positioned(self.cached_position.0),
                sdl2::video::WindowPos::Positioned(self.cached_position.1),
            );
        } else {
            self.cached_size = self.window.size();
            self.cached_position = self.window.position();
            self.window.set_fullscreen(FullscreenType::Desktop).ok();
        }

        self.fullscreen = !self.fullscreen;
    }
```

Reply:
This doesn't work on windows because the app crashes when you manually swap to another window.

The original PR was a hackyish attempt to get the desired functionality without the fullscreen. I will special case it to windows though.

Reply:
@Kethku yeah i think special casing windows to fuzz out this behavior would be the right call

Reply:
How bout now. Does it work on your side now?

Reply:
@Kethku yup, excellant